runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/BeamFnMapTaskExecutor.java

```diff
@@ -307,10 +344,17 @@ void updateProgress() {
           grpcWriteOperation.abortWait();
         }
 
-        BeamFnApi.Metrics metrics = MoreFutures.get(bundleProcessOperation.getMetrics());
+        //TODO: Replace getProcessBundleProgress with getMonitoringInfos when Metrics is deprecated.
```

Comment:
I was thinking on the proper way here and assumed that todo that clears up when the code gets deprecated was the best approach.
1. Creating JIRA that looks like "Cleanup Dataflow Runner Harness" when Metrics are deprecated. Doesn't cover proper workload. So it is not the best idea in my opinion.
2. Assigning username is not beneficial as well, since we do not know when will we completely deprecate Metrics. So it doesn't really differ from regular todo.

Reply:
+1 to tag a JIRA. It's useful to have some additional context for when the TODO inevitably gets stale and needs clarification to be actionable.

The JIRA doesn't have to be explicitly for doing this cleanup. It should track when this TODO is ready to implement or obsolete. In this case, it would make sense to have a JIRA for deprecating the legacy Metrics.

Google's public style guides [offer similar advice](https://google.github.io/styleguide/cppguide.html#TODO_Comments) (we don't follow Google styleguides in Beam, although the advice is useful):

> Use TODO comments for code that is temporary, a short-term solution, or good-enough but not perfect.
> 
> TODOs should include the string TODO in all caps, followed by the name, e-mail address, bug ID, or other identifier of the person or issue with the best context about the problem referenced by the TODO. The main purpose is to have a consistent TODO that can be searched to find out how to get more details upon request. A TODO is not a commitment that the person referenced will fix the problem. Thus when you create a TODO with a name, it is almost always your name that is given.
> 
> `// TODO(kl@gmail.com): Use a "*" here for concatenation operator.`
> `// TODO(Zeke) change this to use relations.`
> `// TODO(bug 12345): remove the "Last visitors" feature`
> 
> If your TODO is of the form "At a future date do something" make sure that you either include a very specific date ("Fix by November 2005") or a very specific event ("Remove this code when all clients can handle XML responses.").